src/types.cpp

```diff
@@ -176,6 +176,7 @@ std::string typestr(Type t)
     case Type::tuple:    return "tuple";    break;
     case Type::timestamp:return "timestamp";break;
     case Type::mac_address: return "mac_address"; break;
+    case Type::data:     return "data";     break;
```

Comment:
I'd really prefer to avoid adding new types if we can help it. I think complexity grows exponentially with every new type.

It seems like Type::data is currently used to signal to map creation code it needs a specific type of map. Could you perhaps do that with a new map constructor?

Reply:
The new constructor seems tricky for this.. but perhaps I can reuse Array or Buffer, I'll check, thanks

Reply:
Thought about it overnight again. It doesn't look like `Type::data` is user visible (ie a user cannot assign a variable to type Type::data). So IMO it should not be a part of the type system.